opbeat/contrib/django/middleware/__init__.py

```diff
@@ -152,15 +152,13 @@ def process_view(self, request, view_func, view_args, view_kwargs):
     def process_response(self, request, response):
         try:
             if hasattr(response, 'status_code'):
-                if getattr(request, '_opbeat_view_func', False):
+                # check if _opbeat_transaction_name is set
+                if hasattr(request, '_opbeat_transaction_name'):
+                    view_func = request._opbeat_transaction_name
+                elif getattr(request, '_opbeat_view_func', False):
                     view_func = get_name_from_func(
                         request._opbeat_view_func)
-                else:
-                    view_func = getattr(
-                        request,
-                        '_opbeat_transaction_name',
```

Comment:
We use it to set the name of the transaction if the response isn't generated by a view but a middleware (see https://github.com/opbeat/opbeat_python/commit/d0ff2d177e). This will continue to work. It's never been a public API though.